client/keys/add.go

```diff
@@ -345,7 +323,35 @@ func printStep() {
 type NewKeyBody struct {
 	Name     string `json:"name"`
 	Password string `json:"password"`
-	Seed     string `json:"seed"`
+	Mnemonic string `json:"mnemonic"`
+	Account  int    `json:"account,string,omitempty"`
+	Index    int    `json:"index,string,omitempty"`
+}
+
+// RecoverKeyBody is recover key request REST body
+type RecoverKeyBody struct {
+	Password string `json:"password"`
+	Mnemonic string `json:"mnemonic"`
+	Account  int    `json:"account,string,omitempty"`
+	Index    int    `json:"index,string,omitempty"`
+}
+
+// function to just create a new seed to display in the UI before actually persisting it in the keybase
+func generateMnemonic(algo keys.SigningAlgo) string {
+	kb := client.MockKeyBase()
+	pass := throwAwayPassword
+	name := "inmemorykey"
+	_, seed, _ := kb.CreateMnemonic(name, keys.English, pass, algo)
+	return seed
+}
+
+func checkErrorREST(w http.ResponseWriter, httpErr int, err error) bool {
```

Comment:
We have a function for this: `WriteErrorResponse`: https://godoc.org/github.com/cosmos/cosmos-sdk/client/utils#WriteErrorResponse

Reply:
..though I really like your take on it (check VS write). How about dropping a `CheckAndWriteErrorResponse()` wrapper in there?